docs/api/assert.md

```diff
@@ -0,0 +1,1960 @@
+
+test('assert.notProperty', () => {
+  assert.notProperty({ tea: { green: 'matcha' } })
+})
+```
+
+## propertyVal
+
+- **Type:** `<T, V>(object: T, property: string, value: V, message?: string) => void`
+
+Asserts that `object` has a direct or inherited property named by `property` with a value given by `value`. Uses a strict equality check (===).
+
+```ts
+import { assert, test } from 'vitest'
+
+test('assert.notPropertyVal', () => {
+  assert.propertyVal({ tea: 'is good' })
+})
+```
+
+## notPropertyVal
+
+- **Type:** `<T, V>(object: T, property: string, value: V, message?: string) => void`
+
+Asserts that `object` does not have a direct or inherited property named by `property` with a value given by `value`. Uses a strict equality check (===).
+
+```ts
+import { assert, test } from 'vitest'
+
+test('assert.notPropertyVal', () => {
+  assert.notPropertyVal({ tea: 'is good' })
+  assert.notPropertyVal({ tea: 'is good' })
+})
+```
+
+## deepPropertyVal
+
+- **Type:** `<T, V>(object: T, property: string, value: V, message?: string) => void`
+
+Asserts that `object` has a direct or inherited property named by `property` with a value given by `value`. Uses a deep equality check.
+
+```ts
+import { assert, test } from 'vitest'
+
+test('assert.deepPropertyVal', () => {
+  assert.deepPropertyVal({ tea: { green: 'matcha' } }, 'tea', { green: 'matcha' })
+})
+```
+
+## notDeepPropertyVal
+
+- **Type:** `<T, V>(object: T, property: string, value: V, message?: string) => void`
+
+Asserts that `object` does not have a direct or inherited property named by `property` with a value given by `value`. Uses a deep equality check.
+
+```ts
+import { assert, test } from 'vitest'
+
+test('assert.deepPropertyVal', () => {
+  assert.notDeepPropertyVal({ tea: { green: 'matcha' } }, 'tea', { black: 'matcha' })
+  assert.notDeepPropertyVal({ tea: { green: 'matcha' } }, 'tea', { green: 'oolong' })
+  assert.notDeepPropertyVal({ tea: { green: 'matcha' } }, 'coffee', { green: 'matcha' })
+})
+```
+
+## nestedProperty
+
+- **Type:** `<T>(object: T, property: string, message?: string) => void`
+
+Asserts that `object` has a direct or inherited property named by `property`, which can be a string using dot- and bracket-notation for nested reference.
+
+```ts
+import { assert, test } from 'vitest'
+
+test('assert.deepPropertyVal', () => {
+  assert.nestedProperty({ tea: { green: 'matcha' } }, 'tea.green')
+})
+```
+
+## notNestedProperty
+
+- **Type:** `<T>(object: T, property: string, message?: string) => void`
+
+Asserts that `object` does not have a direct or inherited property named by `property`, which can be a string using dot- and bracket-notation for nested reference.
+
+```ts
+import { assert, test } from 'vitest'
+
+test('assert.deepPropertyVal', () => {
+  assert.notNestedProperty({ tea: { green: 'matcha' } }, 'tea.oolong')
+})
+```
+
+## nestedPropertyVal
+
+- **Type:** `<T>(object: T, property: string, value: any, message?: string) => void`
```

Comment:
is value required? example doesn't show it

Reply:
Yes, the type definition is correct, but the value was omitted from the example 

so i add value in example again